Automattic/delphin
# Add public folder and reorganize bundles

This pull request is a follow-up of https://github.com/Automattic/delphin/pull/33 that moves all assets as well as the client bundle into a `public` directory. It also moves the server bundle into `server/build` since this file should not be served to the client. From a client perspective we now have the following clean structure for resources:

```
/favicon.ico
/images/background.png
/styles/reset.css
/scripts/bundle.js
```

This pull request also fixes a few other things in the process. For example it:
- Adds props validation to JSX components
- Better handles empty queries in the domain search
- Adds a new script to clean dependencies and any file resulting from a build
#### Testing instructions
1. Run `git checkout add/public-folder` and start your server
2. Clear your browser's cache or use the incognito mode
3. Open the [`Home` page](http://localhost:1337/)
4. Check that everything works as before
5. Check that an empty query does not trigger an API call and clear the list of suggestions
#### Reviews
